src/middlewared/middlewared/plugins/account_/privilege.py

```diff
@@ -334,10 +340,16 @@ async def privileges_for_groups(self, groups_key, group_ids):
     @private
     async def compose_privilege(self, privileges):
         compose = {
+            'roles': set(),
             'allowlist': [],
             'web_shell': False,
         }
         for privilege in privileges:
+            for role in privilege['roles']:
+                compose['roles'] |= self.middleware.role_manager.roles_for_fole(role)
```

Comment:
`roles_for_fole` is back
```suggestion
                compose['roles'] |= self.middleware.role_manager.roles_for_fole(role)
```